tests/monitoring/monitoring_service/test_crash.py

```diff
@@ -0,0 +1,125 @@
+import os
+from typing import List
+from unittest.mock import Mock
+
+from monitoring_service.events import (
+    ActionMonitoringTriggeredEvent,
+    Event,
+    ReceiveChannelOpenedEvent,
+    UpdatedHeadBlockEvent,
+)
+from monitoring_service.service import MonitoringService
+
+
+class MockBlockchainListener:
+
+    def __init__(self, block_events: List[List[Event]]):
+        self.block_events = block_events
+
+    def get_events(self, chain_state, to_block: int):
+        from_block = chain_state.latest_known_block + 1
+        blocks = self.block_events[from_block:to_block + 1]
+        events = [ev for block in blocks for ev in block]  # flatten
+        return chain_state, events
+
+
+def test_crash(
+    monitoring_service: MonitoringService,  # adds stake in ServiceRegistry
+    web3,
+    contracts_manager,
+    server_private_key,
+    token_network_registry_contract,
+    monitoring_service_contract,
+    get_random_address,
+    tmpdir,
+    generate_raiden_clients,
+    token_network,
+):
+    """ Process blocks and compare results with/without crash
+
+    A somewhat meaninful crash handling is simulated by not including the
+    UpdatedHeadBlockEvent in every block.
+    """
+    channel_identifier = 3
+    c1, c2 = generate_raiden_clients(2)
+    monitor_request = c2.get_monitor_request(
+        balance_proof=c1.get_balance_proof(
+            channel_id=channel_identifier,
+            nonce=1,
+            additional_hash='0x11',
+            transferred_amount=2,
+            locked_amount=0,
+            locksroot='0x00',
+        ),
+        reward_amount=0,
+    )
+
+    events = [
```

Comment:
Why is this a list of a list? Do you plan to extend it to have multiple items in each sublist? On the other hand it gets flattened in `MockBlockchainListener` anyways.

Reply:
I constructed it to be a list of blocks, so that each block can have multiple events. This might come in handy when a problem only occurs when two events happen in the same block. It's a bit over-engineered for the current use case, but it's also somehow nice that it's similar to the structure of a real blockchain.